vault/resource_aws_auth_backend_roletag_blacklist_test.go

```diff
@@ -0,0 +1,206 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSAuthBackendRoleTagBlacklist_basic(t *testing.T) {
+	backend := acctest.RandomWithPrefix("aws")
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckAWSAuthBackendRoleTagBlacklistDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAuthBackendRoleTagBlacklistConfig_basic(backend),
+				Check:  testAccAWSAuthBackendRoleTagBlacklistCheck_attrs(backend),
+			},
+		},
+	})
+}
+
+func TestAccAWSAuthBackendRoleTagBlacklist_updated(t *testing.T) {
+	backend := acctest.RandomWithPrefix("aws")
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckAWSAuthBackendRoleTagBlacklistDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAuthBackendRoleTagBlacklistConfig_basic(backend),
+				Check:  testAccAWSAuthBackendRoleTagBlacklistCheck_attrs(backend),
+			},
+			{
+				Config: testAccAWSAuthBackendRoleTagBlacklistConfig_updated(backend),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("vault_aws_auth_backend_roletag_blacklist.test",
+						"backend", backend),
+					resource.TestCheckResourceAttr("vault_aws_auth_backend_roletag_blacklist.test",
+						"safety_buffer", "3600"),
+					resource.TestCheckResourceAttr("vault_aws_auth_backend_roletag_blacklist.test",
+						"disable_periodic_tidy", "true"),
+				),
+			},
+			{
+				Config: testAccAWSAuthBackendRoleTagBlacklistConfig_updated2(backend),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("vault_aws_auth_backend_roletag_blacklist.test",
+						"backend", backend),
+					resource.TestCheckResourceAttr("vault_aws_auth_backend_roletag_blacklist.test",
+						"safety_buffer", "3600"),
+					resource.TestCheckResourceAttr("vault_aws_auth_backend_roletag_blacklist.test",
+						"disable_periodic_tidy", "false"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAuthBackendRoleTagBlacklistDestroy(s *terraform.State) error {
+	client := testProvider.Meta().(*api.Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "vault_aws_auth_backend_roletag_blacklist" {
+			continue
+		}
+		secret, err := client.Logical().Read(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("Error checking for AWS Auth Backend roletag blacklist %q: %s", rs.Primary.ID, err)
+		}
+		if secret != nil {
+			return fmt.Errorf("AWS auth backend roletag blacklist %q still exists", rs.Primary.ID)
+		}
+	}
+	return nil
+}
+
+func testAccAWSAuthBackendRoleTagBlacklistConfig_basic(backend string) string {
+	return fmt.Sprintf(`
+resource "vault_auth_backend" "aws" {
+  path = "%s"
+  type = "aws"
+}
+
+resource "vault_aws_auth_backend_roletag_blacklist" "test" {
+  backend = "${vault_auth_backend.aws.path}"
+  safety_buffer = 8600
```

Comment:
Nitpick: We can reduce the number of these test configurations if we parameterize some of these items like `safety_buffer` (it also makes it obvious why the `resource.TestCheckResourceAttr()` values are what they are)